ts/features/fci/saga/networking/handleCreateFilledDocument.ts

```diff
@@ -37,3 +66,57 @@ export function* handleCreateFilledDocument(
     yield* put(fciLoadQtspFilledDocument.failure(getNetworkError(e)));
   }
 }
+
+/**
+ * Handle the FCI polling saga
+ * This saga is used to poll the QTSP filled_document to check if the
+ * document is ready to be downloaded by the user. The polling is done
+ * every POLLING_FREQ_TIMEOUT seconds and the polling is stopped when
+ * the document is ready or when the polling is stopped if
+ * the polling time exceeds POLLING_TIME_THRESHOLD.
+ */
+export function* watchFciPollSaga(
+  qtspFilledDocumentUrl: FilledDocumentDetailView["filled_document_url"]
+) {
+  while (true) {
+    try {
+      const response = yield* call(fetch, qtspFilledDocumentUrl);
+      const responseStatus = response.status;
+      if (responseStatus === 200) {
+        yield* put(
+          fciPollFilledDocument.success({
+            isReady: true
+          })
+        );
+        yield* put(fciCancelPollingFilledDocument());
+      }
+      yield* delay(POLLING_FREQ_TIMEOUT);
+    } catch (e) {
+      yield* put(fciPollFilledDocument.failure(getNetworkError(e)));
+      yield* put(fciCancelPollingFilledDocument());
+    } finally {
+      if (yield* cancelled()) {
+        const filledDocumentSelector: ReturnType<
+          typeof fciPollFilledDocumentReadySelector
+        > = yield* select(fciPollFilledDocumentReadySelector);
+        if (!filledDocumentSelector) {
+          yield* put(
+            fciPollFilledDocument.failure(
+              getNetworkError(new Error("Polling cancelled"))
+            )
+          );
```

Comment:
@francescopersico the delay in the yielded race when stopped call the cancel task like all others. Here I need to check the **isReady** value in filledDocument selector to call the failure action get a potError used in the FciQtspClausesScreen to manage the error component. What do you think about it?